### PR TITLE
Volokitin

### PR DESCRIPTION
**Pull request by 1CC1**
Project done without tests

### CODE REVIEW

File: src/main/java/ru/javarush/volokitin/questdelta/controller/GameServlet.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    private final GameService gameService = GameService.INSTANCE;
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        Long curQuestionID = (Long) Optional
+                .ofNullable(req
+                        .getSession()
+                        .getAttribute("curQuestionID"))
+                .orElse(0L);
+        QuestionDTO questionDTO = gameService.getQuestion(curQuestionID);
```
Comment (demologin) [inline src/main/java/ru/javarush/volokitin/questdelta/controller/GameServlet.java:28]: Optional need FROM service. TO service better send only raw data

File: src/main/java/ru/javarush/volokitin/questdelta/controller/QuestServlet.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import ru.javarush.volokitin.questdelta.service.QuestionService;
+import ru.javarush.volokitin.questdelta.util.JSP;
+
+import java.io.IOException;
+
+@WebServlet({"", "/quest"})
```
Comment (demologin) [inline src/main/java/ru/javarush/volokitin/questdelta/controller/QuestServlet.java:14]: may be need key(s)

File: src/main/java/ru/javarush/volokitin/questdelta/controller/UserServlet.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        Optional<String> newGameOptional = Common.getRequestParameter(req, "new_game");
+        if (newGameOptional.isPresent()) {
+            req.getSession().setAttribute("curQuestionID", 0L);
+            req.getSession().setAttribute("correctAnswersCount", 0);
+            req.getSession().removeAttribute("user");
+        }
+
+        Optional<String> optionalUser = Optional.ofNullable((String) req.getSession().getAttribute("user"));
```
Comment (demologin) [inline src/main/java/ru/javarush/volokitin/questdelta/controller/UserServlet.java:25]: just if may be better?

File: src/main/java/ru/javarush/volokitin/questdelta/repository/AnswerRepository.java [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+import ru.javarush.volokitin.questdelta.entity.Answer;
+
+public class AnswerRepository extends AbstractRepository<Answer> {
+    private static final AnswerRepository answerRepository = new AnswerRepository();
+    public static AnswerRepository get() {
+        return answerRepository;
+    }
+
+    private AnswerRepository() {
+        map.put(0L, Answer.with()
```
Comment (demologin) [inline src/main/java/ru/javarush/volokitin/questdelta/repository/AnswerRepository.java:12]: very hard init

File: src/main/java/ru/javarush/volokitin/questdelta/repository/QuestionRepository.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+public class QuestionRepository extends AbstractRepository<Question> {
+    private static final QuestionRepository questionRepository = new QuestionRepository();
+    public static QuestionRepository get() {
+        return questionRepository;
+    }
+
+    private QuestionRepository() {
+        map.put(0L, Question.with()
+                .id(0L)
+                .text("Лямбда-выражение преобразовывается компилятором в")
```
Comment (demologin) [inline src/main/java/ru/javarush/volokitin/questdelta/repository/QuestionRepository.java:16]: very hard too

File: src/main/java/ru/javarush/volokitin/questdelta/util/Common.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
+package ru.javarush.volokitin.questdelta.util;
+
+import jakarta.servlet.http.HttpServletRequest;
+
+import java.util.Optional;
+
+public class Common {
```
Comment (demologin) [inline src/main/java/ru/javarush/volokitin/questdelta/util/Common.java:7]: private constructor?

**Review by demologin — APPROVED** [review]
ok

File: src/main/java/ru/javarush/volokitin/questdelta/util/Common.java (hunk 6 above)
Comment (1CC1) [reply src/main/java/ru/javarush/volokitin/questdelta/util/Common.java:7]: didn't understand you :(

File: src/main/java/ru/javarush/volokitin/questdelta/repository/AnswerRepository.java (hunk 4 above)
Comment (1CC1) [reply src/main/java/ru/javarush/volokitin/questdelta/repository/AnswerRepository.java:12]: Are you suggest to replace it with reading from a file?

**Pull request merged**